### ISSUE TITLE
Draggable annotations: documentation and bugreport

### ISSUE THREAD

**Issue by Phlya** [matplotlib/matplotlib#5839]
Hi, I just found out somewhere on the web that annotations can be set to be draggable by just doing:

    plt.annotate(...).draggable()

This is absolutely amazing! But I don't think this is documented anywhere, while it should be (or did I miss it?).

At the same time the behaviour is not perfect: often after dragging one annotation when trying to drag another, the first one would be dragged in the same direction (e.g. I drag the second along x axis, the first one does the same). Sometimes I would get even three annotations moving when I drag one of them.

**Comment by tacaswell** [comment]
It is very loosely documented: http://matplotlib.org/examples/pylab_examples/annotation_demo3.html

Add that it is not in the API docs...

**Comment by Phlya** [comment]
True, if you carefully look into that example you can realize it...

**Comment by tacaswell** [comment]
I agree that it should be much better documented and no one can be blamed for missing it as is.

I only found it by git grep on the source

**Comment by Phlya** [comment]
What do you think of the bug? Or is it not a bug?

**Comment by tacaswell** [comment]
That is probably a bug, it would be worth having a look at the draggable selection logic.  @blink1073 has been doing some very interesting / cool work with selection areas, it would be interesting to see if you could re-use any of that to make draggable annotations work better?

**Comment by WeatherGod** [comment]
I am not entirely certain, but I *think* annotations became draggable by
accident via the draggable legend feature. So I guess that that code never
really got fully tested in situations with multiple annotations. Some
example code would be useful. We can then see if it really is a bug or a
feature.

**Comment by Phlya** [comment]
Example code:

    import numpy as np
    import matplotlib.pyplot as plt
    x = np.arange(10)
    y = x**2
    text = [str(i) for i in zip(x, y)]
    plt.scatter(x, y)
    for a, b, c in zip(x, y, text):
        plt.annotate(c, xy=(a, b), xytext=(a, b+2), arrowprops=dict(arrowstyle="->", color='r', lw=0.5)).draggable()
    plt.show()

Try dragging the annotations. Happens randomly, I think, or at least I haven't noticed a pattern.

**Comment by twmacro** [comment]
I've noticed this behavior too. It does seem mostly random, but I've been able to get repeatable behavior doing this:
   - add two draggable annotations
   - move the second
   - move the first --> the second will move with it

Oddly, if you type a command (I run in IPython with interactive on), it seems to break that link. With more annotations, I noticed that many can be linked and typing a command doesn't break all of them.

One other thing I noticed is that the second step above causes a new `._transform` member of the first annotation to be created (I only compared id's of all `.__dict__` members ... before and after).

**Comment by anntzer** [comment]
I think the bug was fixed in #10776 (somewhat indirectly); the documentation request remains.

**Comment by dstansby** [comment]
Turns out this is the curse of private base classes again

**Comment by QuLogic** [comment]
This is unfortunately still undocumented.
